### PR TITLE
Add github actions support

### PR DESCRIPTION
**Pull request by johnor**
Perhaps try this instead of #17? Looks quite good actually.

See output and how it works here: https://github.com/johnor/stdlib2/pull/1

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          else
+            sudo apt-get install -y clang-${{ matrix.version }} g++-multilib
+            echo "::set-env name=CC::clang-${{ matrix.version }}"
+            echo "::set-env name=CXX::clang++-${{ matrix.version }}"
+          fi
+
+      - name: Install (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          python -m pip install ninja --upgrade
```
Comment (robinlinden) [inline .github/workflows/ci.yaml]: No fixed versions for Windows? :P Other than that I think we should give this a try over Travis.

Comment (johnor) [reply .github/workflows/ci.yaml]: Ah, missed that, I will fix

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Install (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          python -m pip install cmake==3.14.4 ninja==1.9.0.post1 --upgrade
+
+      - name: Build and test
+        run: |
+          ninja --version
+          cmake --version
+          cmake -S . -B build ${{ matrix.cmake-args }}
```
Comment (robinlinden) [inline .github/workflows/ci.yaml]: No `-GNinja`? Sorry, missed this the first review.

Comment (johnor) [reply .github/workflows/ci.yaml]: Ah, it is not used now at all. I can remove it for now, and then we add more jobs in a follow up PR with ninja, what do you think?

Comment (johnor) [reply .github/workflows/ci.yaml]: We probably want to have jobs with both ninja and visual studio as generators

**Pull request merged**